### PR TITLE
Update subaru integration documentation

### PR DESCRIPTION
**Pull request by G-Two**
## Proposed change

- Add documentation for new lock platform
- Minor edits for clarity

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/52852

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/subaru.markdown [hunk 1]
```diff
@@ -30,6 +32,19 @@ Currently, this integration only provides the Sensor platform. Subaru has deploy
...
+In addition to the telematics generational differences, there are two levels of STARLINK subscriptions, "Safety Plus" and "Security Plus". All remote services (such as locks) require a "Security Plus" level subscription.
+
+{% include integrations/config_flow.md %}
+
+<p class='note'>
+If your account includes multiple vehicles, the same PIN will be used for all vehicles. Ensure that you have configured all vehicles in your account to have the same PIN.
+</p>
+
+## Platforms
+### Sensor
```
Comment (frenck) [inline source/_integrations/subaru.markdown]:
```suggestion
## Sensors
```

File: source/_integrations/subaru.markdown [hunk 2]
```diff
@@ -44,10 +59,11 @@ Currently, this integration only provides the Sensor platform. Subaru has deploy
 
 \* Gen 1 odometer only updates every 500 miles <br>
 
-{% include integrations/config_flow.md %}
+### Lock
```
Comment (frenck) [inline source/_integrations/subaru.markdown]:
```suggestion
## Lock
```

File: source/_integrations/subaru.markdown [hunk 3]
```diff
@@ -11,14 +12,15 @@ ha_codeowners:
   - '@G-Two'
 ha_domain: subaru
 ha_platforms:
+  - lock
   - sensor
 ---
 
-The Subaru integration retrieves information provided by Subaru connected vehicle services.  Before using this integration, you must first register and have login credentials to [MySubaru](https://www.mysubaru.com).
+This integration retrieves vehicle information and actuates remote services provided by [Subaru STARLINK](https://www.subaru.com/engineering/starlink/safety-security.html) (currently only available in USA and Canada).
```
Comment (frenck) [inline source/_integrations/subaru.markdown]:
```suggestion
This integration retrieves vehicle information and actuates remote services provided by [Subaru STARLINK](https://www.subaru.com/engineering/starlink/safety-security.html) (currently only available in USA and Canada).
```

File: source/_integrations/subaru.markdown [hunk 4]
```diff
@@ -11,14 +12,15 @@ ha_codeowners:
...
+This integration retrieves vehicle information and actuates remote services provided by [Subaru STARLINK](https://www.subaru.com/engineering/starlink/safety-security.html) (currently only available in USA and Canada).  
 
-This integration requires an active vehicle subscription to the [Subaru STARLINK](https://www.subaru.com/engineering/starlink/safety-security.html) service (available in USA and Canada).
+This integration requires a telematics equipped Subaru and an active vehicle subscription to the Subaru STARLINK service. Before using this integration, you must first register and have login credentials to [MySubaru](https://www.mysubaru.com).
 
-Currently, this integration only provides the Sensor platform. Subaru has deployed two generations of telematics, Gen 1 and Gen 2. The latter offers much more vehicle information. Use the tables below to determine which sensors are available to your vehicle.
+Subaru has deployed two generations of telematics, Gen 1 and Gen 2. The latter offers more vehicle sensor information and additional remote capabilities. Use the table below to determine your vehicle's telematics generation and capabilities.
```
Comment (frenck) [inline source/_integrations/subaru.markdown]:
```suggestion
Subaru has deployed two generations of telematics, Gen 1 and Gen 2. The latter offers more vehicle sensor information and additional remote capabilities. Use the table below to determine your vehicle's telematics generation and capabilities.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks, @G-Two 👍 

Left some really small comments.

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

File: source/_integrations/subaru.markdown [hunk 5]
```diff
@@ -30,6 +32,18 @@ Currently, this integration only provides the Sensor platform. Subaru has deploy
...
+
+{% include integrations/config_flow.md %}
+
+<p class='note'>
+If your account includes multiple vehicles, the same PIN will be used for all vehicles. Ensure that you have configured all vehicles in your account to have the same PIN.
+</p>
+
+## Sensors
+
+Available sensors will vary by model, year, and subscription type. The integration will add all supported sensors for your vehicle. Sensor data is usually only updated when the vehicle is turned off, unless the [polling option](#options) is enabled.
```
Comment (bdraco) [inline source/_integrations/subaru.markdown]:
```suggestion
Available sensors will vary by model, year, and subscription type. The integration will add all supported sensors for your vehicle. Sensor data is usually only updated when the vehicle is turned off unless the [polling option](#options) is enabled.
```

**Review by bdraco — APPROVED** [review]

**Review by frenck — APPROVED** [review]

**Pull request merged**